src/include/migraphx/generate.hpp

```diff
@@ -37,6 +37,12 @@ constexpr T normalize(unsigned long z)
     return z % max;
 }
 
+template <>
```

Comment:
This should probably be `template<class T, MIGRAPHX_REQUIRES(std::is_same<T, bool>{})>`.